<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>862</width>
    <height>707</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>3D-Viewer v2.0</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QMainWindow {
	background: rgb(152, 152, 152)
}
QRadioButton {
	font: 300 14pt &quot;Ubuntu Light&quot;;
}</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="styleSheet">
    <string notr="true">QSpinBox {
	font: 500 14pt &quot;Ubuntu Light&quot;;
}
QGroupBox {
	font: 500 20pt &quot;Ubuntu Light&quot;;
    color: black;
}
/*QLabel*/
QLabel {
	font: 500 18pt &quot;Ubuntu Light&quot;;
    color: black;
}
/*QRadioButton*/
QRadioButton {
	font: 500 18pt &quot;Ubuntu Light&quot;;
    color: black;
}
/* QPushButton */
QPushButton {
	border: 2px solid transparent;
	font: 300 18pt &quot;Ubuntu Light&quot;;
    color: white;
	background: black;
}
QPushButton:hover {
    border:none;
}
QPushButton:pressed {
    background:rgb(145, 48, 0);
}

/* QSlider */
QSlider::groove:horizontal {
    border: 1px solid black;
    height: 8px; 
    background:rgb(145, 48, 0);
    margin: 2px 0;
}

QSlider::handle:horizontal {
	background: black;
    width: 18px;
    margin: -2px 0;
    border-radius: 3px;
}</string>
   </property>
   <layout class="QGridLayout" name="gridLayout_4">
    <item row="1" column="1">
     <widget class="QGroupBox" name="moveGroupBox">
      <property name="title">
       <string>Move</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="2" column="1">
        <widget class="QSlider" name="sliderMoveZ">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>-50</number>
         </property>
         <property name="maximum">
          <number>50</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QSlider" name="sliderMoveY">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>-50</number>
         </property>
         <property name="maximum">
          <number>50</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="labelMoveX">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>X</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSlider" name="sliderMoveX">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>-50</number>
         </property>
         <property name="maximum">
          <number>50</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="labelMoveY">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Y</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="labelMoveZ">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Z</string>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QSpinBox" name="spinBoxMoveX">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="buttonSymbols">
          <enum>QAbstractSpinBox::PlusMinus</enum>
         </property>
         <property name="minimum">
          <number>-50</number>
         </property>
         <property name="maximum">
          <number>50</number>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QSpinBox" name="spinBoxMoveY">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="buttonSymbols">
          <enum>QAbstractSpinBox::PlusMinus</enum>
         </property>
         <property name="minimum">
          <number>-50</number>
         </property>
         <property name="maximum">
          <number>50</number>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QSpinBox" name="spinBoxMoveZ">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="buttonSymbols">
          <enum>QAbstractSpinBox::PlusMinus</enum>
         </property>
         <property name="minimum">
          <number>-50</number>
         </property>
         <property name="maximum">
          <number>50</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="3" column="1">
     <widget class="QGroupBox" name="scaleGroupBox">
      <property name="title">
       <string>Scale</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <widget class="QSlider" name="sliderScale">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>-100</number>
         </property>
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSpinBox" name="spinBoxScale">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="buttonSymbols">
          <enum>QAbstractSpinBox::PlusMinus</enum>
         </property>
         <property name="minimum">
          <number>-100</number>
         </property>
         <property name="maximum">
          <number>100</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QGroupBox" name="rotateGroupBox">
      <property name="title">
       <string>Rotate</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="1" column="0">
        <widget class="QLabel" name="labelRotateY">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Y</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QSlider" name="sliderRotateY">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>-180</number>
         </property>
         <property name="maximum">
          <number>180</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="labelRotateX">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>X</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSlider" name="sliderRotateX">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>-180</number>
         </property>
         <property name="maximum">
          <number>180</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QSlider" name="sliderRotateZ">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>-180</number>
         </property>
         <property name="maximum">
          <number>180</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="labelRotateZ">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Z</string>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QSpinBox" name="spinBoxRotateX">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="buttonSymbols">
          <enum>QAbstractSpinBox::PlusMinus</enum>
         </property>
         <property name="minimum">
          <number>-180</number>
         </property>
         <property name="maximum">
          <number>180</number>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QSpinBox" name="spinBoxRotateY">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="buttonSymbols">
          <enum>QAbstractSpinBox::PlusMinus</enum>
         </property>
         <property name="minimum">
          <number>-180</number>
         </property>
         <property name="maximum">
          <number>180</number>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QSpinBox" name="spinBoxRotateZ">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="buttonSymbols">
          <enum>QAbstractSpinBox::PlusMinus</enum>
         </property>
         <property name="minimum">
          <number>-180</number>
         </property>
         <property name="maximum">
          <number>180</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="s21::FigureRenderer" name="axisView" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>130</width>
        <height>120</height>
       </size>
      </property>
      <property name="mouseTracking">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="0" column="0" rowspan="5">
     <widget class="s21::FigureRenderer" name="modelView" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="mouseTracking">
       <bool>true</bool>
      </property>
      <property name="styleSheet">
       <string notr="true">/*border: 2px solid transparent;
border-radius: 10px;
padding: 30px;
margin: 30px;*/</string>
      </property>
     </widget>
    </item>
    <item row="4" column="1">
     <widget class="QPushButton" name="buttonResetAffine">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string>reset</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>862</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Ubuntu Light</family>
     <pointsize>14</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="font">
     <font>
      <family>Ubuntu Light</family>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen_file"/>
    <addaction name="actionTake_a_screenshot"/>
    <addaction name="actionMake_a_GIF"/>
   </widget>
   <widget class="QMenu" name="menuEdges">
    <property name="font">
     <font>
      <family>Ubuntu Light</family>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="title">
     <string>Edges</string>
    </property>
    <widget class="QMenu" name="edgesColor">
     <property name="font">
      <font>
       <family>Ubuntu Light</family>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="title">
      <string>Color</string>
     </property>
    </widget>
    <widget class="QMenu" name="edgesType">
     <property name="font">
      <font>
       <family>Ubuntu Light</family>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="title">
      <string>Type</string>
     </property>
     <addaction name="actionSolid"/>
     <addaction name="actionDashed"/>
    </widget>
    <widget class="QMenu" name="edgesThickness">
     <property name="font">
      <font>
       <family>Ubuntu Light</family>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="title">
      <string>Thickness</string>
     </property>
    </widget>
    <addaction name="edgesColor"/>
    <addaction name="edgesThickness"/>
    <addaction name="edgesType"/>
   </widget>
   <widget class="QMenu" name="menuVertices">
    <property name="font">
     <font>
      <family>Ubuntu Light</family>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="title">
     <string>Vertices</string>
    </property>
    <widget class="QMenu" name="verticesColor">
     <property name="font">
      <font>
       <family>Ubuntu Light</family>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="title">
      <string>Color</string>
     </property>
    </widget>
    <widget class="QMenu" name="verticesThickness">
     <property name="font">
      <font>
       <family>Ubuntu Light</family>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="title">
      <string>Thickness</string>
     </property>
    </widget>
    <widget class="QMenu" name="verticesType">
     <property name="font">
      <font>
       <family>Ubuntu Light</family>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="title">
      <string>Type</string>
     </property>
     <addaction name="actionNone"/>
     <addaction name="actionCircle"/>
     <addaction name="actionSquare"/>
    </widget>
    <addaction name="verticesColor"/>
    <addaction name="verticesThickness"/>
    <addaction name="verticesType"/>
   </widget>
   <widget class="QMenu" name="menuBackground">
    <property name="font">
     <font>
      <family>Ubuntu Light</family>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="title">
     <string>Background</string>
    </property>
    <widget class="QMenu" name="backgroundColor">
     <property name="title">
      <string>Color</string>
     </property>
    </widget>
    <addaction name="backgroundColor"/>
   </widget>
   <widget class="QMenu" name="menuProjection">
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="title">
     <string>Projection</string>
    </property>
    <widget class="QMenu" name="projectionMode">
     <property name="title">
      <string>Mode</string>
     </property>
     <addaction name="actionParallel"/>
     <addaction name="actionCentral"/>
    </widget>
    <addaction name="projectionMode"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdges"/>
   <addaction name="menuVertices"/>
   <addaction name="menuBackground"/>
   <addaction name="menuProjection"/>
  </widget>
  <action name="actionOpen_file">
   <property name="text">
    <string>Open file</string>
   </property>
   <property name="font">
    <font>
     <family>Ubuntu Light</family>
     <pointsize>14</pointsize>
    </font>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSolid">
   <property name="text">
    <string>Solid</string>
   </property>
  </action>
  <action name="actionDashed">
   <property name="text">
    <string>Dashed</string>
   </property>
  </action>
  <action name="actionNone">
   <property name="text">
    <string>None</string>
   </property>
   <property name="font">
    <font>
     <family>Ubuntu Light</family>
     <pointsize>14</pointsize>
    </font>
   </property>
  </action>
  <action name="actionCircle">
   <property name="text">
    <string>Circle</string>
   </property>
   <property name="font">
    <font>
     <family>Ubuntu Light</family>
     <pointsize>14</pointsize>
    </font>
   </property>
  </action>
  <action name="actionSquare">
   <property name="text">
    <string>Square</string>
   </property>
   <property name="font">
    <font>
     <family>Ubuntu Light</family>
     <pointsize>14</pointsize>
    </font>
   </property>
  </action>
  <action name="actionTake_a_screenshot">
   <property name="text">
    <string>Take a screenshot</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+T</string>
   </property>
  </action>
  <action name="actionMake_a_GIF">
   <property name="text">
    <string>Make a GIF</string>
   </property>
   <property name="shortcut">
    <string>Shift+G</string>
   </property>
  </action>
  <action name="actionReset">
   <property name="text">
    <string>Reset</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
  <action name="actionParallel">
   <property name="text">
    <string>Parallel</string>
   </property>
   <property name="font">
    <font>
     <family>Ubuntu</family>
     <pointsize>14</pointsize>
    </font>
   </property>
  </action>
  <action name="actionCentral">
   <property name="text">
    <string>Central</string>
   </property>
   <property name="font">
    <font>
     <family>Ubuntu</family>
     <pointsize>14</pointsize>
    </font>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>s21::FigureRenderer</class>
   <extends>QWidget</extends>
   <header>view/figurerenderer.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
